homeassistant/components/stream/core.py

```diff
@@ -386,17 +386,28 @@ async def handle(
         raise NotImplementedError()
 
 
-TRANSFORM_IMAGE_FUNCTION = (
-    lambda image: image,  # Unused
-    lambda image: image,  # No transform
-    lambda image: np.fliplr(image).copy(),  # Mirror
-    lambda image: np.rot90(image, 2).copy(),  # Rotate 180
-    lambda image: np.flipud(image).copy(),  # Flip
-    lambda image: np.flipud(np.rot90(image)).copy(),  # Rotate left and flip
-    lambda image: np.rot90(image).copy(),  # Rotate left
-    lambda image: np.flipud(np.rot90(image, -1)).copy(),  # Rotate right and flip
-    lambda image: np.rot90(image, -1).copy(),  # Rotate right
-)
+def _transform_image(image: np.ndarray, orientation: int) -> np.ndarray:
+    """Transform the image to a given orientation using numpy."""
+
+    if orientation < 2 or orientation > 8:  # Unused/identity orientations
+        return image
+
+    # Keep import here so that we can import stream integration without installing reqs
```

Comment:
So just to be clear, why are we adding this additional complexity?

Reply:
Agree with Frenck.

Reply:
It sounds like a huge waste of energy, time, bandwidth, and disk space to expect all 593 packages listed in `requirements_test_all.txt` to be installed (in e.g. codespaces, devcontainers, ...) just to be able to test a single component's changes (which is my use case).

This, with #81791 and adding `recorder` requirements to `requirements_test.txt` (which happens in #81795) enables being able to do that.

Of course, `numpy` could just be made a hard dependency for `stream` and I could add `stream`'s requirements to `requirements_test.txt` in #81795, and that would work too...

Reply:
>  sounds like a huge waste of energy, time, bandwidth, and disk space to expect all 593 packages listed in requirements_test_all.txt to be installed (in e.g. codespaces, devcontainers, ...) just to be able to test a single component's changes (which is my use case).

Codespaces are pre-built, so that should not be a factor. Anyways, testing a single component and its dependencies depends on what you are testing. Meaning, not installing @ full becomes unpredictable, regardless of these changes. Therefore, I don't think we should add exceptions/complexity to support these cases.

Reply:
> Hm, isn't this what the strict dependencies/requirements manifest system tries to avoid? 

No?

The point is: This PR added tailored changes and add complexity that enabled the testing of "some" integrations in a non-fully set-up development environment. To me, this is solving a specific and limited use case, that, above all, is inconsistent (as it will not always work). Therefore, I'm against this change, and I think we should close it for that reason.

../Frenck

Reply:
Fair. Merging #81841 (which just notes that `numpy` is a hard requirement for `stream`, which it is) will close this one.